### PR TITLE
Added testcases for the C#11 nameof scope change

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**

### CODE REVIEW

**Comment by costin-zaharia-sonarsource** [comment]
We might want to add a test case for `AvoidExcessiveClassCoupling` as well. It seems that we look for `nameof` only in invocations.

Attributes have an argument list but they are not considered invocations: 
image

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]
I would add a test case also for:
- `ControllingPermissions.cs`
```
[Obsolete(nameof(httpContext.User))] // Noncompliant, FN
void HttpContextUser(HttpContext httpContext)
{
}
```
This happens because the `PropertyAccessTracker` subscribes to `SimpleMemberAccessExpression`.
- there are other trackers as well that use this kind of subscription. Could you please check them and the rule that are using them too? (as far as I can tell the rules that use the `MatchSetter` are not a problem and also the ones that look for symbols as the symbol will be null)

**Comment by gregory-paidis-sonarsource** [comment]
> We might want to add a test case for AvoidExcessiveClassCoupling as well. It seems that we look for nameof only in invocations. 

I do not believe this is necessary. Since `AvoidExcessiveClassCoupling` checks for method arguments, a potential type dependency would have already been counted from the `ParameterSyntax` node on the method declaration, which is what the nameof would reference.

**Comment by gregory-paidis-sonarsource** [comment]
> I would add a test case also for:
> 
> * `ControllingPermissions.cs`

After looking into this, it seems like we do not raise for the `HttpContext.User`, but the reason for not raising is not the usage of the nameof as it should work, but because of how the type gets resolved. 

For HttpContext, we resolve it to this known type:

`System_Web_HttpContext = new("System.Web.HttpContext");`

This is the where the type resides in [.net framework](https://learn.microsoft.com/en-us/dotnet/api/system.web.httpcontext?view=netframework-4.8), but not where it resides in [.net core](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httpcontext?view=aspnetcore-7.0)

They seem to have moved it from `System.Web` to `Microsoft.AspNetCore.Http`, meaning that even if you normally access `httpContext.User` it fails to raise the issue:

```csharp
var user = httpContext.User; // Compliant, False Negative (C#11)
```

If you add the following to the PropertyTracker, it marks both the `httpContext.User` and the `nameof(httpContext.User)` as Noncompliant:

```csharp
Microsoft_AspNetCore_Http_HttpContext = new("Microsoft.AspNetCore.Http.HttpContext");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ControllingPermissions.CSharp11.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Threading;
+using Microsoft.AspNetCore.Http;
+
+[Obsolete(nameof(httpContext.User))] // Compliant, because HttpContext class was moved from System.Web to Microsoft.AspNetCore.Http and we do not track it
+void HttpContextUser(HttpContext httpContext)
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ControllingPermissions.CSharp11.cs:6]: The table is not populated for this rule. You need to specify also "Implement" to fix the FNs and the FP below.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/EmptyNullableValueAccessTest.cs [hunk 2]
```diff
@@ -26,33 +26,43 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class EmptyNullableValueAccessTest
     {
-        private static readonly DiagnosticDescriptor[] OnlyDiagnostics = new[] { EmptyNullableValueAccess.S3655 };
+        private static readonly VerifierBuilder Builder = new VerifierBuilder<SymbolicExecutionRunner>()
+          .WithOnlyDiagnostics(EmptyNullableValueAccess.S3655)
+          .WithBasePath(@"SymbolicExecution\Sonar")
+          .WithConcurrentAnalysis(false);
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/EmptyNullableValueAccessTest.cs:32]:
```suggestion
        private readonly VerifierBuilder builder = new VerifierBuilder<SymbolicExecutionRunner>()
            .WithOnlyDiagnostics(EmptyNullableValueAccess.S3655)
            .WithBasePath(@"SymbolicExecution\Sonar")
            .WithConcurrentAnalysis(false);
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.CSharp11.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        private void Argument_Reassigned(string argument) // Noncompliant
+//                                       ^^^^^^^^^^^^^^^
+        {
+            argument = "So Long, and Thanks for All the Fish";
+        }
+
+        [Obsolete(nameof(argument))]
+        private void Argument_UsedInAttributeByNameOf(string argument) // Compliant, because this rule ignores methods that have attributes, not because of the nameof.
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.CSharp11.cs:20]:
```suggestion
        private void Argument_UsedInAttributeByNameOf(string argument) // Compliant, methods with attributes are ignored
```

also below

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**